### PR TITLE
Enhance base64 JSVM handlers

### PR DESCRIPTION
**Pull request by matiasinsaurralde**
For #1808

### CODE REVIEW

File: mw_js_plugin.go [hunk 1]
```diff
@@ -422,12 +422,50 @@ func (j *JSVM) LoadTykJSApi() {
...
+			out, err = base64.RawStdEncoding.DecodeString(in)
+			if err != nil {
+				log.WithFields(logrus.Fields{
+					"prefix": "jsvm",
+				}).Error("Failed to base64 decode: ", err)
+				return otto.Value{}
+			}
+		}
+		returnVal, err := j.VM.ToValue(string(out))
 		if err != nil {
```
Comment (asoorm) [inline mw_js_plugin.go]: can we remove this error check - seems like duplication?

**Comment by matiasinsaurralde** [comment]
Just added tests, included a few ones that deal with the encoded/decoded JWT payload we were experimenting with 👍

**Comment by asoorm** [comment]
@ilijabojanovic for 2.7.2?

**Comment by ilijabojanovic** [comment]
@asoorm yes. From this morning all finished PR's going in to 2.7.2 @buger .

**Comment by ilijabojanovic** [comment]
@buger @matiasinsaurralde 

On latest master: 04972c3735a35a35a69db76caf8a0c6bad0a5616
I still see errors with base64 decoding

```
[Aug 24 16:53:01]  INFO jsvm-logmsg: Virtual Test running type=log-msg
[Aug 24 16:53:01]  INFO jsvm-logmsg: full JWT: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c type=log-msg
[Aug 24 16:53:01]  INFO jsvm-logmsg: header: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9 type=log-msg
[Aug 24 16:53:01]  INFO jsvm-logmsg: payload: eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ type=log-msg
[Aug 24 16:53:01]  INFO jsvm-logmsg: signature: SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c type=log-msg
[Aug 24 16:53:01] ERROR jsvm: Failed to base64 decode: illegal base64 data at input byte 34
[Aug 24 16:53:01]  INFO jsvm-logmsg: headerDecoded: {"alg":"HS256","typ":"JWT"} type=log-msg
[Aug 24 16:53:01]  INFO jsvm-logmsg: payloadDecoded: {"sub":"1234567890","name":"John Doe","iat":1516239022} type=log-msg
[Aug 24 16:53:01]  INFO jsvm-logmsg: signatureDecoded: undefined type=log-msg
```

**Comment by matiasinsaurralde** [comment]
Can you share the JS code you're using to test this?

**Comment by buger** [comment]
@ilijabojanovic bump ^

**Comment by ilijabojanovic** [comment]
@buger i we sent him data over slack DM few days ago :) 
@matiasinsaurralde i we used same example from task

**Pull request merged**